x-pack/plugins/monitoring/init.js

```diff
@@ -27,16 +27,22 @@ export const init = (monitoringPlugin, server) => {
   xpackMainPlugin.status.once('green', async () => {
     const config = server.config();
     const uiEnabled = config.get('xpack.monitoring.ui.enabled');
+    const kibanaCollectionEnabled = config.get('xpack.monitoring.kibana.collection.enabled');
+    const mainXpackInfo = server.plugins.xpack_main.info;
+    const mainMonitoring = mainXpackInfo.feature('monitoring');
+    const monitoringBulkEnabled = mainMonitoring.isAvailable() && mainMonitoring.isEnabled();
 
     if (uiEnabled) {
       await instantiateClient(server); // Instantiate the dedicated ES client
       await initMonitoringXpackInfo(server); // Route handlers depend on this for xpackInfo
       await requireUIRoutes(server);
     }
 
-    if (config.get('xpack.monitoring.kibana.collection.enabled')) {
-      const collectorSet = createCollectorSet(monitoringPlugin.kbnServer, server); // instantiate an object for collecting/sending metrics and usage stats
-      server.expose('collectorSet', collectorSet); // expose the collector set object on the server. other plugins will call statsCollectors.register(collector) to define their own collection
+    const { collectorSet, bulkUploader } = initKibanaMonitoring(monitoringPlugin.kbnServer, server); // instantiate objects for collecting stats and uploading stats
+    server.expose('collectorSet', collectorSet); // expose the collectorSet service
+
+    if (kibanaCollectionEnabled && monitoringBulkEnabled) {
+      bulkUploader.start(); // start the internal uploader for collected metrics
```

Comment:
Per the previous comments on this area, I still need to handle stopping/restarting the uploader when elasticsearch plugin goes red/back to green